otx/algorithms/common/tasks/base_task.py

```diff
@@ -331,3 +332,7 @@ def config(self):
     @config.setter
     def config(self, config: Dict[Any, Any]):
         self._config = config
+
+    @staticmethod
+    def _get_adaptive_num_workers():
+        return min(multiprocessing.cpu_count() // torch.cuda.device_count(), 8)  # max available num_workers is 8
```

Comment:
I'm very curious why max available num_workers is 8? Is it empirically set? And is multiprocessing.cpu_count() returns available num cpus?

Reply:
I'm very curious why max available num_workers is 8?
=> Because current maximum value available to set is 8
does multiprocessing.cpu_count() returns available num cpus?
=> Yes, it returns number of cores of cpu